### PR TITLE
node 20; oidc-ts-client 3.x; remove itwin- 3.x packages

### PR DESCRIPTION
**Pull request by ben-polinsky**
- Update ci and node types to 20.x. Not only is 20.x LTS, but it is required for oidc-ts-client@3.x which relies on the browser's crypto. Node>18.x automatically adds `crypto` to `globalThis`. 
- update oidc-ts-client to 3.x
- remove support for itwin 3.x packages

closes: #246

### CODE REVIEW

File: packages/browser/package.json [hunk 1]
```diff
@@ -56,6 +56,6 @@
     "typescript": "~5.3.3"
   },
   "peerDependencies": {
-    "@itwin/core-bentley": "^3.3.0 || ^4.0.0"
+    "@itwin/core-bentley": "^4.0.0"
```
Comment (aruniverse) [inline packages/browser/package.json:59]: are we going to make this a major bump?

File: packages/browser/package.json [hunk 2]
```diff
@@ -31,17 +31,17 @@
...
-    "oidc-client-ts": "^2.4.0"
+    "@itwin/core-common": "^4.0.0",
+    "oidc-client-ts": "^3.0.1"
   },
   "devDependencies": {
-    "@itwin/build-tools": "^4.6.0-dev.27",
-    "@itwin/core-bentley": "^3.7.0",
-    "@itwin/eslint-plugin": "^4.0.2",
+    "@itwin/build-tools": "^4.7.1",
+    "@itwin/core-bentley": "^4.7.1",
```
Comment (aruniverse) [inline packages/browser/package.json:39]: does core-bentley need to be in lockstep with the other core pkgs?

File: packages/browser/package.json (hunk 1 above)
Comment (ben-polinsky) [reply packages/browser/package.json:59]: Yep that is the idea

File: packages/browser/package.json (hunk 2 above)
Comment (ben-polinsky) [reply packages/browser/package.json:39]: Are you suggesting they should both be 4.7.0? Thought for dev deps our policy was to bump to latest compatible with deps.

**Comment by ben-polinsky** [comment]
Don't know that it's worth moving to pure esm until core is there...

**Comment by ben-polinsky** [comment]
> Don't know that it's worth moving to pure esm until core is there...

After discussion with @aruniverse, we'll pick this back up closer to core@5 release. We'll go full ESM at that point as well.

**Pull request closed without merging**